e2e/etcd_test.go

```diff
@@ -433,9 +433,9 @@ func (ep *etcdProcess) Stop() error {
 }
 
 func (ep *etcdProcess) waitReady() error {
-	readyStrs := []string{"enabled capabilities for version", "published"}
+	readyStrs := []string{"starting server...", "published"}
```

Comment:
OK, I see kubernetes has full-fledged versioned e2e conformance tests; that versioning functionality is definitely intentional. Forward rolling testing (e.g., use 3.0.x tests against 3.1) seems reasonable for e2e, integration, and clientv3/integration tests using etcd binaries independent from test fixtures, but it'll still need some thought to get it right.